main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1052,6 +1052,13 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
           val sp1 = leftMostSourcePosition(exp1)
           WeededAst.Expression.FixpointEntails(e1, e2, mkSL(sp1, sp2))
       }
+
+    case ParsedAst.Expression.FixpointFold(sp1, pred, init, f, constraints, sp2) =>
+      val loc = mkSL(sp1, sp2)
+      mapN(visitExp(init), visitExp(f), visitExp(constraints)) {
+        case (e1, e2, e3) =>
+          WeededAst.Expression.FixpointFold(WeededAst.PredicateWithParam(pred, /* TODO: not sure about this argument */ WeededAst.Expression.Unit(loc)), e1, e2, e3, loc)
```

Comment:
In the paper, we describe "IndexedPredicates" which is a way of writing Foo<1>(...) and Foo<2>() and have them be two different predicates, but with the same types.

Reply:
Thanks for the clarification! Support for this is added in commit 362e8ff